packages/js/product-editor/src/components/details-name-block/edit.tsx

```diff
@@ -2,36 +2,175 @@
  * External dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { createElement, createInterpolateElement } from '@wordpress/element';
-import { TextControl } from '@woocommerce/components';
+import {
+	createElement,
+	Fragment,
+	createInterpolateElement,
+	useState,
+} from '@wordpress/element';
+
 import { useBlockProps } from '@wordpress/block-editor';
+import { cleanForSlug } from '@wordpress/url';
+import { useSelect, useDispatch } from '@wordpress/data';
+import {
+	PRODUCTS_STORE_NAME,
+	WCDataSelector,
+	Product,
+} from '@woocommerce/data';
+import {
+	Button,
+	BaseControl,
+	// @ts-expect-error `__experimentalInputControl` does exist.
+	__experimentalInputControl as InputControl,
+} from '@wordpress/components';
 // eslint-disable-next-line @typescript-eslint/ban-ts-comment
 // @ts-ignore No types for this exist yet.
 // eslint-disable-next-line @woocommerce/dependency-group
-import { useEntityProp } from '@wordpress/core-data';
+import { useEntityProp, useEntityId } from '@wordpress/core-data';
+
+/**
+ * Internal dependencies
+ */
+import { AUTO_DRAFT_NAME } from '../../utils';
+import { EditProductLinkModal } from '../edit-product-link-modal';
+import { useValidation } from '../../hooks/use-validation';
+import { useProductHelper } from '../../hooks/use-product-helper';
 
 export function Edit() {
 	const blockProps = useBlockProps();
-	const [ name, setName ] = useEntityProp( 'postType', 'product', 'name' );
+
+	const { updateProductWithStatus } = useProductHelper();
```

Comment:
Hmm does the `core` store also allow us to do something like this?
Using `updateProductWithStatus` would technically be fine, but it does use a totally different data store, or could we just use `editEntityRecord` without directly saving it?

Reply:
@louwie17 Yeah, it felt a little odd to me but I made the assumption that we wanted to retain the previous functionality of saving the product link immediately from the modal "Save" button. It would be simpler just to edit the entity as we are elsewhere, but the change isn't persisted immediately of course. 

If we wanted to change this behavior perhaps we would change the wording on the button as well, although I'm personally alright with the current behavior as well.

Reply:
Thanks for checking with @jarekmorawski, in the mean time, I would be fine leaving this as is, and changing this as a follow up if we decide to change the behaviour.
Although instead of `updateProductWithStatus` could you still make use of `saveEntityRecord`?

Reply:
Yup, for some reason I was thinking `saveEntityRecord` would only save _new_ records, but providing the ID allows it to save without disturbing the `editedEntityRecord` (still getting down this API). Done in [`1541bb0` (#37612)](https://github.com/woocommerce/woocommerce/pull/37612/commits/1541bb04e783c49bb771ad3f85b1b29f5ef4dadc)